gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/reactive/handlers/api/SyncApiReactor.java

```diff
@@ -52,47 +61,117 @@ public class SyncApiReactor extends AbstractLifecycleComponent<ReactorHandler> i
         this.defaultInvoker = defaultInvoker;
         this.resourceLifecycleManager = resourceLifecycleManager;
         this.policyManager = policyManager;
         this.groupLifecycleManager = groupLifecycleManager;
+
+        final PolicyChainFactory policyChainFactory = new DefaultPolicyChainFactory(api.getId(), policyManager);
+
+        this.platformFlowChain =
+            new FlowChain("Platform", FlowResolverFactory.forPlatform(api, organizationManager), platformPolicyChainFactory);
+        this.apiPlanFlowChain = new FlowChain("Api Plan", FlowResolverFactory.forApiPlan(api), policyChainFactory);
+        this.apiFlowChain = new FlowChain("Api", FlowResolverFactory.forApi(api), policyChainFactory);
     }
 
     @Override
     public Completable handle(Request request, Response response) {
         final RequestExecutionContext ctx = ctxFactory.createRequestContext(request, response);
 
+        // Prepare attributes and metrics before handling the request.
+        prepareContextAttributes(ctx);
+        prepareMetrics(ctx);
+
+        return handleRequest(ctx);
+    }
+
+    private void prepareContextAttributes(RequestExecutionContext ctx) {
         ctx.setAttribute(ExecutionContext.ATTR_CONTEXT_PATH, ctx.request().contextPath());
         ctx.setAttribute(ExecutionContext.ATTR_API, api.getId());
         ctx.setAttribute(ExecutionContext.ATTR_API_DEPLOYED_AT, api.getDeployedAt().getTime());
         ctx.setAttribute(ExecutionContext.ATTR_INVOKER, defaultInvoker);
         ctx.setAttribute(ExecutionContext.ATTR_ORGANIZATION, api.getOrganizationId());
         ctx.setAttribute(ExecutionContext.ATTR_ENVIRONMENT, api.getEnvironmentId());
+    }
 
-        // Prepare request metrics
-        request.metrics().setApi(api.getId());
-        request.metrics().setPath(request.pathInfo());
+    private void prepareMetrics(RequestExecutionContext ctx) {
+        final Request request = ctx.request();
+        final Metrics metrics = request.metrics();
 
-        ctx.request().metrics().setApiResponseTimeMs(System.currentTimeMillis());
+        metrics.setApi(api.getId());
+        metrics.setPath(request.pathInfo());
+        metrics.setApiResponseTimeMs(System.currentTimeMillis());
+    }
 
-        // FIXME: security chain with plan resolution.
-        ctx.setAttribute(io.gravitee.gateway.api.ExecutionContext.ATTR_PLAN, "cd593616-5a15-4972-9936-165a15c972be");
+    private Completable handleRequest(RequestExecutionContext ctx) {
+        // Execute request flows (platform, security, api plan, api).
+        return executeChain(ctx, platformFlowChain, REQUEST)
+            .andThen(continueChain(ctx, apiPlanFlowChain, REQUEST))
+            .andThen(continueChain(ctx, apiFlowChain, REQUEST))
+            // All request flows have been executed. Invokes backend.
+            .andThen(invokeBackend(ctx))
+            // Execute response flows (api plan, api, platform).
+            .andThen(continueChain(ctx, apiPlanFlowChain, RESPONSE))
```

Comment:
I am not sure to understand the order. Should'nt be 

1. platform
2. plan
3. api
4. backend
5. api
6. plan
7. platform ?